gobblin-metrics/src/main/java/gobblin/metrics/event/sla/SlaEventSubmitter.java

```diff
@@ -0,0 +1,128 @@
+package gobblin.metrics.event.sla;
+
+import java.util.Map;
+import java.util.Properties;
+
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.commons.lang.StringUtils;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Predicate;
+import com.google.common.base.Strings;
+import com.google.common.collect.Maps;
+
+import gobblin.metrics.event.EventSubmitter;
+
+/**
+ * A wrapper around the {@link EventSubmitter} which can submit SLA Events.
+ */
+@Builder
+@Slf4j
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class SlaEventSubmitter {
+
+  private EventSubmitter eventSubmitter;
+  private String eventName;
+  private String datasetUrn;
+  private String partition;
+  private String originTimestamp;
+  private String upstreamTimestamp;
+  private String recordCount;
+  private String previousPublishTimestamp;
+  private String dedupeStatus;
+  private String completenessPercentage;
+  private Map<String, String> additionalMetadata;
```

Comment:
can you add `@singular` annotation to this? https://projectlombok.org/features/Builder.html